<launch>
  <arg name="electric_gripper" default="false"/>

  <!-- Start the Sawyer pick and place demo -->
  <node pkg="sorting_demo" type="program.py" name="sorting_demo" >
    <remap from="/joint_states" to="/robot/joint_states"/>
  </node>

  <param name="tip_link_name" value="right_hand_camera_optical" if="$(arg electric_gripper)"/>
  <param name="tip_link_name" value="right_hand_camera" unless="$(arg electric_gripper)"/>


  <node pkg="topic_tools" type="relay" args="/robot/joint_states /joint_states" name="statedup"/>
  <node pkg="sawyer_ik_5d" type="sawyer_ik_5d_node" name="sawyer_ik_5d_node"  output="screen"/>

  <node pkg="intera_interface" type="joint_trajectory_action_server.py" name="trajectory_action_server"/>

  <node pkg="sorting_demo" type="gripper_action_server.py" name="gripper_action_server" output="screen"/>


  <node pkg="tf" type="static_transform_publisher"
        name="camera_broadcaster_optical"
        args="0 0 0 0 -1.5707963267948966 0 right_hand_camera right_hand_camera_optical 100" />


  <include file="$(find sawyer_moveit_config)/launch/sawyer_moveit.launch">
        <arg name="electric_gripper" value="$(arg electric_gripper)"/>
  </include>
</launch>